src/context.js

```diff
@@ -473,7 +473,7 @@ export default class Context {
 					this._queueExposure(experimentName, assignment);
 				}
 
-				if (key in assignment.variables) {
+				if (key in assignment.variables && assignment.eligible) {
```

Comment:
Shouldn't be check of `assigned` here? `eligible` only check if the user is in the traffic allocation split, but the user may also be filtered out by the audience.
I think we should also allow the variant if the `overridden` flag is set, for testing purposes.
So I think the test should be `assigned || overridden` (like in Java)
Can you add a test for this case to validate this assumption?